pkg/services/authz/zanzana/store/store.go

```diff
@@ -31,26 +31,26 @@ func NewStore(cfg *setting.Cfg, logger log.Logger) (storage.OpenFGADatastore, er
 	switch grafanaDBCfg.Type {
 	case migrator.SQLite:
 		connStr := sqliteConnectionString(grafanaDBCfg.ConnectionString)
-		if err := migration.Run(cfg, migrator.SQLite, connStr, assets.EmbedMigrations, assets.SqliteMigrationDir); err != nil {
+		if err := migration.Run(cfg, migrator.SQLite, grafanaDBCfg, assets.EmbedMigrations, assets.SqliteMigrationDir, logger); err != nil {
 			return nil, fmt.Errorf("failed to run migrations: %w", err)
 		}
 
 		return sqlite.New(connStr, zanzanaDBCfg)
 	case migrator.MySQL:
 		// For mysql we need to pass parseTime parameter in connection string
 		connStr := grafanaDBCfg.ConnectionString + "&parseTime=true"
-		if err := migration.Run(cfg, migrator.MySQL, connStr, assets.EmbedMigrations, assets.MySQLMigrationDir); err != nil {
+		if err := migration.Run(cfg, migrator.MySQL, grafanaDBCfg, assets.EmbedMigrations, assets.MySQLMigrationDir, logger); err != nil {
 			return nil, fmt.Errorf("failed to run migrations: %w", err)
 		}
 
 		return mysql.New(connStr, zanzanaDBCfg)
 	case migrator.Postgres:
-		connStr := grafanaDBCfg.ConnectionString
-		if err := migration.Run(cfg, migrator.Postgres, connStr, assets.EmbedMigrations, assets.PostgresMigrationDir); err != nil {
+		// Parse and transform the connection string to the format OpenFGA expects
+		if err := migration.Run(cfg, migrator.Postgres, grafanaDBCfg, assets.EmbedMigrations, assets.PostgresMigrationDir, logger); err != nil {
 			return nil, fmt.Errorf("failed to run migrations: %w", err)
 		}
 
-		return postgres.New(connStr, zanzanaDBCfg)
+		return postgres.New(grafanaDBCfg.ConnectionString, zanzanaDBCfg)
```

Comment:
why do we need to change this?

Reply:
we need to have the "original" connection string for grafana. we constructred the connectionstring for openfga. but grafana could not use that with the migrator inside our new `Run()` method